src/Management/src/Endpoint/Actuators/HttpExchanges/HttpExchangesDiagnosticObserver.cs

```diff
@@ -0,0 +1,202 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Steeltoe.Management.Diagnostics;
+
+namespace Steeltoe.Management.Endpoint.Actuators.HttpExchanges;
+
+internal sealed class HttpExchangesDiagnosticObserver : DiagnosticObserver, IHttpExchangesRepository
+{
+    private const string DiagnosticName = "Microsoft.AspNetCore";
+    private const string DefaultObserverName = "TraceDiagnosticObserver";
+    private const string StopEventName = "Microsoft.AspNetCore.Hosting.HttpRequestIn.Stop";
+
+    internal const string Redacted = "******";
+
+    private static readonly DateTime BaseTime = DateTime.UnixEpoch;
+    private readonly IOptionsMonitor<HttpExchangesEndpointOptions> _optionsMonitor;
+    private readonly ILogger<HttpExchangesDiagnosticObserver> _logger;
+    internal ConcurrentQueue<HttpExchange> Queue { get; } = new();
+
+    public HttpExchangesDiagnosticObserver(IOptionsMonitor<HttpExchangesEndpointOptions> optionsMonitor, ILoggerFactory loggerFactory)
+        : base(DefaultObserverName, DiagnosticName, loggerFactory)
+    {
+        ArgumentNullException.ThrowIfNull(optionsMonitor);
+
+        _optionsMonitor = optionsMonitor;
+        _logger = loggerFactory.CreateLogger<HttpExchangesDiagnosticObserver>();
+    }
+
+    public HttpExchangesResult GetHttpExchanges()
+    {
+        return new HttpExchangesResult(Queue.ToList());
```

Comment:
Redaction should take place at the time the actuator executes. So that users can turn it off in configuration without restarting the app, and then see what they were looking for in the captured exchanges. Otherwise, they'll need to replay the scenario first.